### PR TITLE
New filter component

### PR DESCRIPTION
**Pull request by SMahil**
This PR include new filter component

### CODE REVIEW

**Comment by mindreeper2420** [comment]
@dlabrecq Can you take a look at this?

File: src/app/new-filters/n-filter-fields.component.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<div class="ng-filter-bar">
+  <a class="ng-filter-icon dib" (click)="openFilterList">
+    <i class="fa fa-filter fa-lg"></i>
+  </a>
+  <div class="ng-filter-list filter-listright">
+    <div class="filter-search">
+      <input type="text" class="form-control">
+      <a>
+        <i class="fa fa-lg fa-close"></i>
```
Comment (mindreeper2420) [inline src/app/new-filters/n-filter-fields.component.html]: `fa-lg` should be last

File: src/app/new-filters/n-filter-fields.component.scss [hunk 2]
```diff
@@ -0,0 +1,84 @@
+@import '../../assets/stylesheets/_base';
+
+.ng-filter-bar {
+  position: relative;
+  .ng-filter-icon {
+    padding: 5px 7px;
+    color: #4d5258;
```
Comment (mindreeper2420) [inline src/app/new-filters/n-filter-fields.component.scss]: use PatternFly color variable

File: src/app/new-filters/n-filter-fields.component.scss [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    border: 1px solid;
+    border-color: #bbb;
```
Comment (mindreeper2420) [inline src/app/new-filters/n-filter-fields.component.scss]: use PatternFly color variable

File: src/app/new-filters/n-filter-fields.component.scss [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    position: absolute;
+    top: 100%;
+    left: 0;
+    float: left;
+    min-width: 176px;
+    .filter-search {
+      display: inline-block;
+      width: 100%;
+      border: 1px solid;
+      border-color: #bbb;
```
Comment (mindreeper2420) [inline src/app/new-filters/n-filter-fields.component.scss]: use PatternFly color variable

File: src/app/new-filters/n-filter-fields.component.scss [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        margin: 0;
+        box-shadow: none;
+      }
+      &:hover .ng-filter-list-sec {
+        display: block;
+        animation: fadeIn 0.5s;
+      }
+      .ng-filter-list-sec {
+        position: inherit;
+        //display: block;
```
Comment (mindreeper2420) [inline src/app/new-filters/n-filter-fields.component.scss]: remove

File: src/app/new-filters/n-filter-fields.component.scss [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        top: -1px;
+        margin: 0;
+        margin-left: -1px;
+        overflow: scroll;
+        box-shadow: none;
+      }
+    }
+  }
+  .filter-list-right {
+    position: relative !important;
```
Comment (mindreeper2420) [inline src/app/new-filters/n-filter-fields.component.scss]: remove `!important`

File: src/app/new-filters/n-filter-fields.component.scss [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    right: 0;
+    left: auto;
+    float: right !important;
```
Comment (mindreeper2420) [inline src/app/new-filters/n-filter-fields.component.scss]: remove `!important`

File: src/app/new-filters/n-filter-fields.component.scss [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    .list-wrapper {
+      right: 0;
+      .ng-filter-list-primary {
+        right: 0;
+        float: right !important;
```
Comment (mindreeper2420) [inline src/app/new-filters/n-filter-fields.component.scss]: remove `!important`

File: src/app/new-filters/n-filter.component.scss [hunk 9]
```diff
@@ -0,0 +1,5 @@
+@import '../../assets/stylesheets/_base';
+.filter-pf {
+  a { cursor: pointer; }
+}
+.dropdown-menu { min-width: 176px; }
```
Comment (mindreeper2420) [inline src/app/new-filters/n-filter.component.scss]: with view encapsulation turned off, has it been verified that this change to the width of the class `.dropdown-menu` will not negatively impact the other dropdown menus on any page that this widget is included on?

File: src/app/app.module.ts [hunk 10]
```diff
@@ -23,6 +26,12 @@ import { TreeListExampleModule } from './treelist/examples/treelist-example.modu
     FormsModule,
     HomeModule,
     HttpModule,
+    ListViewExampleModule,
+    NewFilterExampleModule,
+    NotificationExampleModule,
+    RemainingCharsExampleModule,
+    SortExampleModule,
+
```
Comment (mindreeper2420) [inline src/app/app.module.ts]: remove space

File: src/app/home/home.component.html [hunk 11]
```diff
@@ -14,4 +14,9 @@
       <a routerLink="/treelist" routerLinkActive="active">Tree List</a>
     </div>
   </div>
+  <div class="row">
+    <div class="col-xs-12">
```
Comment (mindreeper2420) [inline src/app/home/home.component.html]: is this verified to work at the `xs` size? if not, please set to `col-sm-12` to match the other widgets.

File: src/app/new-filters/examples/n-filter-example.component.html [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+    </div>
+  </div>
+  <div class="row">
+    <div class="col-sm-12">
+      <div class="form-group">
+        <alm-n-filter [config]="filterConfig" [queries]="queryConfig" (onChange)="filterChange($event)" (onFilterKeySelection)="getFilterKeyValues($event)" (typeAhedFilterQueries)="filterQueries($event)"></alm-n-filter>
+      </div>
+    </div>
+  </div>
+  <!--
```
Comment (mindreeper2420) [inline src/app/new-filters/examples/n-filter-example.component.html]: remove code that is commented out

File: src/app/new-filters/examples/n-filter-example.component.ts [hunk 13]
```diff
@@ -0,0 +1,326 @@
...
+import { Router } from '@angular/router';
+
+import { NewFilter } from '../n-filter';
+import { NewFilterConfig } from '../n-filter-config';
+import { NewFilterQuery } from '../n-filter-query';
+import { NewFilterField } from '../n-filter-field';
+import { NewFilterEvent } from '../n-filter-event';
+
+@Component({
+  host: {'class': 'app app-component flex-container in-column-direction flex-grow-1'},
```
Comment (mindreeper2420) [inline src/app/new-filters/examples/n-filter-example.component.ts]: why is all of this here?

File: src/app/new-filters/n-filter-results.component.html [hunk 14]
```diff
@@ -0,0 +1,20 @@
+<div class="filter-pf" *ngIf="config && config.appliedFilters && config.appliedFilters.length > 0">
+  <div class="row toolbar-pf-results">
+    <div class="col-sm-12">
+      <h5 *ngIf="config.resultsCount >= 0">{{config.resultsCount}} Results</h5>
+      <p *ngIf="config.appliedFilters.length > 0">Active filters:</p>
+      <ul class="list-inline">
+        <li *ngFor="let filter of config.appliedFilters">
+          <span class="active-filter label label-info">
+            {{filter.field.title}}: {{filter.value}}
+            <a><span class="pficon pficon-close" (click)="clearFilter(filter)"></span></a>
```
Comment (mindreeper2420) [inline src/app/new-filters/n-filter-results.component.html]: why not just apply the `class` and `(click)` to the `<a>`?

File: src/app/new-filters/n-filter-results.component.html [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+          </span>
+        </li>
+      </ul>
+      <p><a class="clear-filters" (click)="clearAllFilters()" *ngIf="config.appliedFilters.length > 0">Clear All Filters</a></p>
```
Comment (mindreeper2420) [inline src/app/new-filters/n-filter-results.component.html]: place the `<a>` on a new line

**Comment by mindreeper2420** [comment]
@SMahil @michaelkleinhenz what are the differences between this and the PatternFly-ng filter and toolbar components that are already included in OpenShift.io?

https://rawgit.com/patternfly/patternfly-ng/master-dist/dist-demo/#/toolbar

File: src/app/home/home.component.html (hunk 11 above)
Comment (SMahil) [inline src/app/home/home.component.html]: This is only for reference(in ngx-widgets) how widgets are working.

File: src/app/new-filters/examples/n-filter-example.component.ts (hunk 13 above)
Comment (SMahil) [inline src/app/new-filters/examples/n-filter-example.component.ts]: I have this from previous filters. It seems ngx-widgets is still using flex

File: src/app/new-filters/n-filter-results.component.html (hunk 14 above)
Comment (SMahil) [inline src/app/new-filters/n-filter-results.component.html]: For icons its good to use spans. Also it is same as per Patternfly

File: src/app/app-routing.module.ts [hunk 16]
```diff
@@ -5,6 +5,7 @@ import { EmptyStateExampleComponent } from './emptystate/examples/emptystate-exa
 import { FilterExampleComponent } from './filters/examples/filter-example.component';
 import { HomeComponent } from './home/home.component';
 import { ListViewExampleComponent } from './listview/examples/listview-example.component';
+import { NewFilterExampleComponent } from './new-filters/examples/n-filter-example.component';
```
Comment (joshuawilson) [inline src/app/app-routing.module.ts]: We should not name something "new" as it will not always be "new"

**Comment by michaelkleinhenz** [comment]
@mindreeper2420 it has a different UX as requested by Samir. To be honest, I did expect that the design we got is in line with UX and Patternfly. We should really sort this out.

**Comment by dlabrecq** [comment]
This looks a lot like the patternfly-ng filter we just removed from ngx-widgets. Before I dive too deep, it would help if someone shared the UX explaining the differences? If there is something the patternfly-ng filter does not support, we should consider adding it there. It's too difficult supporting similar components in multiple repos.

**Comment by joshuawilson** [comment]
I agree with @dlabrecq, please provide the UX so we can see if this requirement can be carried in patternfly-ng.

**Comment by mindreeper2420** [comment]
@SMahil is there any update on this? If not, can it be closed in favor of creating a story to add this to patternfly-ng?

**Comment by dlabrecq** [comment]
Are there any new features other than "saved filters"? It turns out that patternfly-ng has implemented a saved filters feature.

The PR is going thru minor design tweaks, but you can take a look here:
https://github.com/patternfly/patternfly-ng/pull/155

**Comment by SMahil** [comment]
@mindreeper2420 Yes we can close this

**Pull request closed without merging**